carbon-design-system/carbon
# feat(text-input): add inline option for TextInput (#5915)

Closes #5915

Changes TextInput react component to allow for inline variant
and adds required styles. Updates TextInput Storybook page
to include inline variant in Knobs tab.

#### Changelog

**Changed**

- TextInput React component (packages/react/src/components/TextInput/TextInput.js)
- TextInput styles (packages/components/src/components/text-input/_text-input.scss)
- TextInput Storybook page (packages/react/src/components/TextInput/TextInput-story.js)

#### Testing / Reviewing

1. fetch/checkout branch and run `storybook` locally
2. open Storybook in browser and navigate to TextInput > Default page
3. click on Knobs tab
4. scroll to bottom of Knobs and toggle on `Inline variant (inline)`
5. observe the inline variant of input
6. toggle on `Show form validation UI (invalid)` to see invalid state of input

![image](https://user-images.githubusercontent.com/5007314/84326854-e0c31680-ab4b-11ea-9aa9-0bc5901d45cc.png)

![image](https://user-images.githubusercontent.com/5007314/84326910-ffc1a880-ab4b-11ea-9866-a19645dfd9b8.png)
